builders/scripts/build_debian_geany.sh

```diff
@@ -0,0 +1,182 @@
+
+parse_geany_version() {
+	log "Parse Geany version"
+	GEANY_VERSION=$(sed -n -E -e 's/^AC_INIT\(\[Geany\], \[(.+)\],/\1/p' ${GEANY_BUILD_DIR}/configure.ac)
+	GEANY_GIT_REVISION=$(cd ${GEANY_BUILD_DIR} && git rev-parse --short --revs-only HEAD 2>/dev/null || true)
+	PACKAGE_VERSION="${GEANY_VERSION}-1~$(date '+%Y%m%d')git${GEANY_GIT_REVISION}"
+}
+
+
+run_autogen_sh() {
+	log "Run ./autogen.sh"
+	cd ${GEANY_BUILD_DIR}
+	# run ./autogen.sh as the Debian package won't run ./autogen.sh for us
+	NOCONFIGURE=1 ./autogen.sh
+}
+
+create_tarball_for_debuild() {
+	log "Create source tarball"
+	cd ${GEANY_BUILD_DIR}
+	# create a source tarball, keep .git included on purpose for Geany GIT build detection
+	tar --transform "s,^,geany-plugins-${GEANY_VERSION}/,S" \
+		--create \
+		--gzip \
+		--file "../geany_${GEANY_VERSION}.orig.tar.gz" \
+		.
+}
+
+
+git_clone_debian_geany() {
+	log "Clone Geany Debian repository"
+	git clone --depth 1 ${GEANY_DEBIAN_REPOSITORY} ${GEANY_DEBIAN_DIR}
+	# move debian package files to Geany GIT checkout
+	mv ${GEANY_DEBIAN_DIR}/debian ${GEANY_BUILD_DIR}
+}
+
+
+install_build_dependencies() {
+	log "Install build dependencies"
+	cd ${GEANY_BUILD_DIR}
+	apt-get --assume-yes update
+	apt-get --assume-yes dist-upgrade
+	mk-build-deps \
+		--install \
+		--remove \
+		--tool "apt-get -o Debug::pkgProblemResolver=yes --no-install-recommends --assume-yes"
+	# mk-build-deps leave the following files which then break 'debuild' later on
+	rm -f geany-build-deps_*-?_amd64.buildinfo geany-build-deps_*-?_amd64.changes
+}
+
+
+add_changelog_entry() {
+	log "Add ChangeLog entry for Debian package"
+	cd ${GEANY_BUILD_DIR}
+	dch \
+		--urgency low \
+		--distribution unstable \
+		--newversion "${PACKAGE_VERSION}" \
+		"Autobuilt by CI or nightly build."
+}
+
+
+build_package() {
+	log "Build Debian package"
+	cd ${GEANY_BUILD_DIR}
+
+	debuild -uc -us
+}
+
+
+copy_results() {
+	log "Copy build artefacts"
+
+	BUILD_DIR_DIRNAME=$(dirname ${GEANY_BUILD_DIR})
+
+	rm -rf ${OUTPUT_DIRECTORY}/geany/*
+
+	cp --verbose ${BUILD_DIR_DIRNAME}/geany_* ${OUTPUT_DIRECTORY}/geany
+	cp --verbose ${BUILD_DIR_DIRNAME}/geany-common_* ${OUTPUT_DIRECTORY}/geany
+	cp --verbose ${BUILD_DIR_DIRNAME}/geany-dbgsym_* ${OUTPUT_DIRECTORY}/geany
+}
+
+
+log_and_store_build_environment() {
+	log "Using environment"
+
+	GTK_VERSION=$(pkg-config --modversion gtk+-3.0)
+	GLIB_VERSION=$(pkg-config --modversion glib-2.0)
+	COMPILER_VERSION=$(gcc -dumpfullversion)
+
+	echo "Debian Distribution  : $(grep PRETTY_NAME /etc/os-release | cut -d '=' -f 2)"
+	echo "Geany version        : ${GEANY_VERSION}"
+	echo "Geany GIT revision   : ${GEANY_GIT_REVISION}"
+	echo "GLib version         : ${GLIB_VERSION}"
+	echo "GTK version          : ${GTK_VERSION}"
+	echo "GCC version          : ${COMPILER_VERSION}"
+
+	cat <<EOT > ${OUTPUT_DIRECTORY}/geany/versions.json
+	{
+		"glib_version": "${GLIB_VERSION}",
+		"gtk_version": "${GTK_VERSION}",
+		"gcc_version": "${COMPILER_VERSION}",
+	}
+EOT
```

Comment:
Nothing yet but it might become useful when using the scripts for nightly builds to fill the table on https://www.geany.org/download/nightly-builds/.

But I'm quite undecided yet how to proceed with the nightly builds exactly and the whole Debian package builds scripts are yet to be finalized. For now, I would like to focus on the mingw64 parts to get them integrated into the pipeline.

Reply:
I would add such artifacts only when they became needed with a clear use case.

Reply:
They are gone with the whole Debian code.